pkgs/tools/security/wordlists/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib
+, pkgs
+, runtimeShell
+, symlinkJoin
+, tree
+}:
+
+let
+  _wordlistData =
+    [ { name = "dirbuster"; expressionPath = ./dirbuster.nix; }
+      { name = "nmap"; expressionPath = ./nmap.nix; needsPackage = true; }
+      { name = "rockyou"; expressionPath = ./rockyou.nix; }
+      { name = "seclists"; expressionPath = ./seclists.nix; }
+      { name = "wfuzz"; expressionPath = ./wfuzz.nix; needsPackage = true; }
+    ];
+
+  scopedWordlists = lib.makeScope pkgs.newScope (self: with self;
+    let _packageWithLinkCommandFrom = acc: datum:
+      # Some wordlists are based on an already existing `nixpkgs` package; Pass it as parameter of the expression.
+      let parameters = {}
+        // lib.optionalAttrs ((builtins.elem "needsPackage" (builtins.attrNames datum)) && datum.needsPackage) { ${datum.name} = pkgs.${datum.name}; };
+      in
+        let p = callPackage datum.expressionPath parameters;
+        in acc // { ${datum.name} = p; };
+    in builtins.foldl' _packageWithLinkCommandFrom {} _wordlistData
+  );
+
+in let
+  packageConstructor = wordlistPackages:
+    symlinkJoin rec {
+      pname = "security-wordlists";
+      version = "unstable-2020-11-23";
+
+      name = "${pname}-${version}";
+
+      paths = wordlistPackages;
+
+      dontUnpack = true;
```

Comment:
it should be documented what this line does.

Reply:
I don't really know which line this comment applies to...
If it is about `dontUnpack`, it is documented at https://nixos.org/manual/nixpkgs/stable/#ssec-unpack-phase; If it is about `paths`, then it is a parameter of the `symlinkJoin` function used, and it is documented at https://nixos.org/manual/nixpkgs/stable/#chap-trivial-builders .

Reply:
other `runCommand` invocations (`symlinkJoin` also does this) don't do this, e.g. https://github.com/NixOS/nixpkgs/blob/dfcfe50920c81302be069533514a16d4a48c6715/pkgs/build-support/trivial-builders.nix#L255-L271; therefore I'm not sure if it is really necessary.

Reply:
I tested it without it and couldn't detect a difference in the output, therefore I assume that it isn't necessary.